### PR TITLE
Feature/jsonnet tool

### PR DESCRIPTION
**Pull request by steeling**
Leverage the jsonnet files from feature/jsonnet to create more separation between files assocaited with different kubernetes clusters.

The bulk of the code allows specifying the context in the file itself, such that we can have multiple leaf files. It also adds a diff + confirmation before mutating operations.

For the rest of the tools we migrate generated/namespace to workspace/cluster, and force specifying the cluster as an arg to make-certs and apply-certs. We also give recs to copy minimum.jsonnet into generated/cluster so that each directory is fully self contained, and there is no possibility of pushing to the wrong cluster

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -153,9 +163,19 @@ following differences:
...
+
 1.  You must run ./make-certs.py with the `--ca-cert-to-join` flag as described
     above to use the existing cluster's CA to sign your certificates.
 
+1.  You must share ca.crt with the cluster(s) you are trying to join, and have
+    them apply the new ca.crt, which now contains both your cluster and the
+    original clusters public certs, to enable secure bi-directional
+    communication.
+
+1.  All of the original clusters must then perform a rolling restart of their
```
Comment (charlie-pisuraj) [inline build/README.md]: Can we add a Note on the top that warns the use to create all the certs at once IF they are going to deploy multiple clusters that needs to talk to each other by themselves.

I know it should be incredibly rare for some one to do that but that really saved me so much time and confusion by creating all the certs

File: build/apply-certs.sh [hunk 2]
```diff
@@ -1,35 +1,42 @@
...
 
 # Paths to directories in which to store certificates and generated YAML files.
+CONTEXT="$1"
 DIR="$(pwd)"
 NAMESPACE="dss-main"
-CLIENTS_CERTS_DIR="$DIR/generated/$NAMESPACE/client_certs_dir"
-NODE_CERTS_DIR="$DIR/generated/$NAMESPACE/node_certs_dir"
-CA_KEY_DIR="$DIR/generated/$NAMESPACE/ca_key_dir"
-CA_CRT_DIR="$DIR/generated/$NAMESPACE/ca_certs_dir"
+CLIENTS_CERTS_DIR="$DIR/workdspace/$CONTEXT/client_certs_dir"
```
Comment (charlie-pisuraj) [inline build/apply-certs.sh]: Love this, I actually did something similar locally to handle multiple cluster

Comment (steeling) [reply build/apply-certs.sh]: +1 :)

File: build/README.md (hunk 1 above)
Comment (steeling) [reply build/README.md]: done, added note

File: build/apply-certs.sh [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
+NODE_CERTS_DIR="$DIR/workdspace/$CONTEXT/node_certs_dir"
+CA_KEY_DIR="$DIR/workdspace/$CONTEXT/ca_key_dir"
+CA_CRT_DIR="$DIR/workdspace/$CONTEXT/ca_certs_dir"
```
Comment (charlie-pisuraj) [inline build/apply-certs.sh]: typo workdspace

File: build/README.md [hunk 4]
```diff
@@ -103,13 +98,10 @@ which ever provider you choose.
...
     Ingress needs to be created as a "Global" IP address.
 
-1.  Copy `values.yaml.template` to `values.yaml` and fill in the required fields
-    at the top.
-
 1.  Use the `make-certs.py` script in this directory to create certificates for
     the new CockroachDB cluster:
 
-        ./make-certs.py [--node-address <ADDRESS> <ADDRESS> <ADDRESS> ...]
+        ./make-certs.py <cluster_context> [--node-address <ADDRESS> <ADDRESS> <ADDRESS> ...]
```
Comment (charlie-pisuraj) [inline build/README.md]: --cluster flag needed

File: build/apply-certs.sh (hunk 3 above)
Comment (steeling) [reply build/apply-certs.sh]: done

**Comment by steeling** [comment]
ping on this, can we move this PR forward?

**Comment by charlie-pisuraj** [comment]
looks like test broke and it can't install kubecfg

**Comment by charlie-pisuraj** [comment]
image

**Pull request merged**